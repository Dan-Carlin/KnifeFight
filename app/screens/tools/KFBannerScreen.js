import React from "react";
import { View, StyleSheet } from "react-native";

import Screen from "../../components/Screen";
import Text from "../../components/Text";

const background = require("../../assets/backgrounds/kf_background_xxxhdpi.png");

function KFBannerScreen(props) {
  return (
    <Screen style={styles.screenContainer} background={background}>
      <View style={styles.buttonContainer}>
        <Text>buttonContainer</Text>
      </View>
      <View style={styles.nameContainer}>
        <Text>nameContainer</Text>
      </View>
      <View style={styles.healthContainer}>
        <Text>healthContainer</Text>
      </View>
    </Screen>
  );
}

const styles = StyleSheet.create({
  buttonContainer: {
    alignItems: "center",
    // backgroundColor: "crimson",
    flex: 0.25,
  },
  healthContainer: {
    alignItems: "center",
    // backgroundColor: "limegreen",
    flex: 0.35,
  },
  nameContainer: {
    alignItems: "center",
    // backgroundColor: "skyblue",
    flex: 0.4,
  },
  screenContainer: {
    // backgroundColor: "gold",
    flex: 1,
  },
});

export default KFBannerScreen;
