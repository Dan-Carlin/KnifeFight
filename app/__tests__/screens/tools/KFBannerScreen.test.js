import React from "react";
import renderer from "react-test-renderer";

import KFBannerScreen from "../../../screens/tools/KFBannerScreen";

describe("Banner screen tests", () => {
  it("renders correctly", () => {
    const route = { params: { isAccesibilityModeOn: false } };

    const tree = renderer.create(<KFBannerScreen route={route} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
