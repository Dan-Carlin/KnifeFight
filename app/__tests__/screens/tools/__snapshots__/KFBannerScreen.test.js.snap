// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner screen tests renders Banner screen correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "flex": 1,
      "transform": Array [
        Object {
          "rotate": "180deg",
        },
      ],
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <ViewManagerAdapter_ExpoLinearGradient
    proxiedProperties={
      Object {
        "colors": Array [
          0,
          4293667645,
        ],
        "endPoint": Array [
          0,
          1,
        ],
        "locations": undefined,
        "startPoint": Array [
          0,
          0.5,
        ],
      }
    }
    style={
      Object {
        "height": "100%",
        "opacity": 0.4,
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <SvgMock
    style={
      Object {
        "height": "100%",
        "opacity": 0.7,
        "position": "absolute",
        "width": "100%",
      }
    }
  />
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Array [
        Object {
          "paddingTop": 20,
        },
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 0.25,
        }
      }
    >
      <View
        style={
          Object {
            "height": "100%",
            "transform": Array [
              Object {
                "rotate": "180deg",
              },
            ],
            "width": "20%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "height": "100%",
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <SvgMock />
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.4,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "center",
            "width": "100%",
          }
        }
      >
        <Text
          adjustsFontSizeToFit={true}
          mode="max_lines"
          numberOfLines={1}
          onTextLayout={[Function]}
          style={
            Array [
              Object {
                "color": "#ec2b3d",
                "fontFamily": "default",
                "textAlign": "center",
                "width": "94%",
              },
              Object {
                "fontSize": 200,
              },
            ]
          }
          testID="max-lines"
        >
          Gang Name
        </Text>
        <Text
          style={
            Array [
              Object {
                "color": "#fff",
                "fontFamily": "Avenir",
                "fontSize": 16,
                "width": "100%",
              },
              Object {
                "textAlign": "center",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "color": "#bcbec0",
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
          >
            (The 
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "fontStyle": "italic",
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
          >
            Unstable
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "color": "#bcbec0",
                  "fontWeight": "bold",
                  "textAlign": "center",
                },
              ]
            }
          >
             ones)
          </Text>
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 0.35,
        }
      }
    >
      <View
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 0.4,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#fff",
                  "fontFamily": "Avenir",
                  "fontSize": 16,
                  "width": "100%",
                },
                Object {
                  "textAlign": "center",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontFamily": "Avenir",
                    "fontSize": 16,
                    "width": "100%",
                  },
                  Object {
                    "fontSize": 60,
                    "fontWeight": "bold",
                    "lineHeight": 63,
                  },
                ]
              }
            >
              25
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontFamily": "Avenir",
                    "fontSize": 16,
                    "width": "100%",
                  },
                  Object {
                    "color": "#ec2b3d",
                    "fontFamily": "default",
                    "fontSize": 50,
                  },
                ]
              }
            >
              HP
            </Text>
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 0.6,
            }
          }
        >
          <View
            style={
              Object {
                "borderColor": "#bcbec0",
                "borderWidth": 6,
                "height": "60%",
                "padding": 8,
                "width": "90%",
              }
            }
          >
            <ViewManagerAdapter_ExpoLinearGradient
              proxiedProperties={
                Object {
                  "colors": Array [
                    4288945474,
                    4293667645,
                  ],
                  "endPoint": Array [
                    1,
                    0,
                  ],
                  "locations": undefined,
                  "startPoint": Array [
                    0,
                    0,
                  ],
                }
              }
              style={
                Object {
                  "height": "100%",
                  "width": "100%",
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
